R/population-study-design.R

```diff
@@ -0,0 +1,226 @@
+#' @title addStudyParameters
+#' @param population `Population` object
+#' @param studyDesignFile file name of study design table
+#' @export
+#' @import ospsuite
+addStudyParameters <- function(population, studyDesignFile = NULL) {
+  validateIsOfType(population, "Population")
+  validateIsString(studyDesignFile, nullAllowed = TRUE)
+  if (is.null(studyDesignFile)) {
+    return(invisible())
+  }
+  studyDesign <- loadStudyDesign(studyDesignFile)
+
+  initialTargetValues <- rep(NA, population$count)
+  populationData <- ospsuite::populationAsDataFrame(population)
+
+  for (target in studyDesign$targets) {
+    parameterPath <- target$name
+
+    updatedTargetValues <- updateTargetValues(initialTargetValues, target$values, studyDesign$source, populationData)
+
+    population$setParameterValues(parameterPath, updatedTargetValues)
+  }
+}
+
+#' @title updateTargetValues
+#' @param values initial vector to update
+#' @param targetValues vector of values to be assigned using `sourceExpressions`
+#' @param sourceExpressions study design expressions to be evaluated
+#' @param data population data as data.frame
+#' @export
+#' @import ospsuite
+updateTargetValues <- function(values, targetValues, sourceExpressions, data) {
+  validateIsSameLength(targetValues, sourceExpressions)
+  validateIsOfType(data, "data.frame")
+  validateIsOfLength(values, nrow(data))
+
+  for (sourceIndex in seq_along(sourceExpressions)) {
+    sourceFilter <- eval(sourceExpressions[sourceIndex])
+
+    values[sourceFilter] <- targetValues[sourceIndex]
+  }
+
+  return(values)
+}
+
+#' @title loadStudyDesign
+#' @description Load a StudyDesign object from a file containing a study design table
+#' In this table, Line 1 is path,  Line 2 is unit, Line 3 is type and subsequent lines are values.
+#' The cells for type can include "SOURCE" OR "TARGET". "SOURCE" type can also include subtype "MIN", "MAX" or "EQUALS".
+#' In the current version, unit is expected to be intern units of simulation.

```

Comment:
Do I understand t right: at the moment the units in a design study files are ignored and the values in a design study file must be always in the internal OSP units (e.g. age always in years etc.)?
If so: why?

Reply:
Yes. It is currently ignored.
At the moment, only a simulation object has the knowledge and concept of unit and dimension.

So there can be 2 ways of dealing with unit: 
1) Add the concept in a population object (this would make a lot of changes in the ospsuite package)
2) Include the simulation object to the function addStudyParameter (preferred way)
Use the simulation object to check if added Study Parameter exists in the simulation and that its unit/dimension exist.
If you have an actual example to try that out, I may be able to include that in the current PR.

Reply:
Would it be enough to convert every column to the base unit of its corresponding dimension?
(@msevestre Does ospsuite-R provide such a possibility? )

Then the simulation object would not be required.

Reply:
I see, the function `toBaseUnit` is what you are referring to.
The process would then be, getDimensionFor study design unit, and convert unit toBaseUnit.
I can add that process to fix the unit.
In the end, I still think we'll need the simulation object at some point since the extended population is meant to be passed on the simulation run (so we can check before the run that paths and their units match and are correctly defined).

Reply:
I just tried to implement and it does not work: `toBaseUnit` requires a quantity object which is derived from a simulation object.
I think including the simulation directly is the best solution.
@Yuri05, is it okay if I through an error if TARGET paths do not exist ?

Reply:
*throw

Reply:
I can update the code in toBaseUnit to be similar to what `toUnit` does and support a dimension name as well. In that case, the simulation would not be required

But you would not check that the path exists. 
I have a few updates to do to ospsuite-r so I'll do that today and then you can update the PR